sdk/iot/provisioning/src/az_iot_provisioning_client.c

```diff
@@ -465,6 +467,9 @@ AZ_NODISCARD az_result az_iot_provisioning_client_parse_received_topic_and_paylo
     return AZ_ERROR_IOT_TOPIC_NO_MATCH;
   }
 
+  az_log_write(AZ_LOG_MQTT_RECEIVED_TOPIC, received_topic);
```

Comment:
NOTE: 2 separate calls to az_log_write are NOT guaranteed to be atomic as another thread could also be calling az_log_write at the same time. If you want the 2 log entries to be considered as an atomic unit, YOU must combine the messages into 1 and call az_log_write ONCE.

Reply:
I think it's interesting to keep topics and payload logging separate. The main purpose of the logs is to enable our CSS team to debug - with some clever filters I think we should be fine (I have some sample filters for the .NET SDK here as example: https://github.com/Azure/azure-iot-sdk-csharp/tree/master/tools/CaptureLogs/query).